From ddf1dc3367bedfb6ee65745908d65d13c8361958 Mon Sep 17 00:00:00 2001
From: Alex Marginean <alexandru.marginean@nxp.com>
Date: Thu, 6 Dec 2018 19:07:51 +0200
Subject: [PATCH 393/706] net: mscc: felix: PHY integration fixes

- don't phy_start at probe
- use proper adjust_link

Signed-off-by: Alex Marginean <alexandru.marginean@nxp.com>
(cherry picked from commit 8a4e7d1cb69a875c120f9d029efd96ee1114fd53)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/ethernet/mscc/felix_board.c | 1 -
 drivers/net/ethernet/mscc/ocelot.c      | 2 +-
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/mscc/felix_board.c b/drivers/net/ethernet/mscc/felix_board.c
index 25b79c149b99..325ce11318af 100644
--- a/drivers/net/ethernet/mscc/felix_board.c
+++ b/drivers/net/ethernet/mscc/felix_board.c
@@ -433,7 +433,6 @@ static int felix_ports_init(struct ocelot *ocelot)
 		if (ndev && port <= FELIX_EXT_CPU_PORT_ID)
 			felix_register_xmit_handler(ocelot->ports[port], ndev);
 		/* TODO: check if FL phy require phy_start */
-		phy_start(phydev);
 	}
 
 	return 0;
diff --git a/drivers/net/ethernet/mscc/ocelot.c b/drivers/net/ethernet/mscc/ocelot.c
index f125127dfa9e..16193bde2a58 100644
--- a/drivers/net/ethernet/mscc/ocelot.c
+++ b/drivers/net/ethernet/mscc/ocelot.c
@@ -488,7 +488,7 @@ static int ocelot_port_open(struct net_device *dev)
 			 ANA_PORT_PORT_CFG_PORTID_VAL(port->chip_port),
 			 ANA_PORT_PORT_CFG, port->chip_port);
 
-	err = phy_connect_direct(dev, port->phy, &ocelot_port_adjust_link_dummy,
+	err = phy_connect_direct(dev, port->phy, ocelot->port_adjust_link,
 				 PHY_INTERFACE_MODE_NA);
 	if (err) {
 		netdev_err(dev, "Could not attach to PHY\n");
-- 
2.17.1

